### DISCUSSION TITLE
[@types/react] Ref callback function should be required to accept null argument

### DISCUSSION THREAD

**Discussion by itsjohncs** [DefinitelyTyped/DefinitelyTyped#58464] [Issues with a @types package]
A ref callback will receive `null` as an argument. But it's possible to give React a ref callback that does not accept `null` without a type error.

This code should give an error but it does not ([playground link](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgIilQ3wG4AoczAVwDsNgJa4jMBhFAG04CN0BrABRJOSEElowAXHAASAFQCyAGQAiwAG4BRUeMkBKGRojAAJnADe5ODbgB6O+QC+lNEwDO8EWInwAvHAAPKaaLEiYfhasHNx8aPxOAHyBdiEaiaRAA)).

```ts
import React from "react";

function refCallback(element: HTMLDivElement): void {
    //
}

const element = <div ref={refCallback}></div>;
```

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
Thanks for opening the issue. This actually hides quite a lot of bugs in this repository.

This is currently caused by our bivariance hack. But I can't reproduce why we needed it here. Or maybe it was only required for older TypeScript version.

I opened https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58936 with a fix to get some more eyes on it. Maybe somebody remembers if we still need the bivariance hack.
